frontend/src/metabase/dashboard/actions/auto-wire-parameters/utils.ts

```diff
@@ -120,21 +121,27 @@ export function getAutoWiredMappingsForDashcards(
   }
   return targetDashcardMappings;
 }
-
+// TODO: this function should automatically calculate return type based on the
+// type of dashcard
 export function getParameterMappings(
-  dashcard: QuestionDashboardCard,
+  dashcard: DashboardCard,
   parameter_id: ParameterId,
   card_id: CardId,
   target: ParameterTarget | null,
 ) {
   const isVirtual = isVirtualDashCard(dashcard);
   const isAction = isActionDashCard(dashcard);
 
-  let parameter_mappings = dashcard.parameter_mappings || [];
+  let parameter_mappings: (
+    | DashboardParameterMapping
+    | ActionParametersMapping
+    | VirtualDashCardParameterMapping
+  )[] = dashcard.parameter_mappings || [];
 
   // allow mapping the same parameter to multiple action targets
   if (!isAction) {
     parameter_mappings = parameter_mappings.filter(
+      // @ts-expect-error action and virtual dashcards do not have card_id
       m => m.card_id !== card_id || m.parameter_id !== parameter_id,
```

Comment:
why not check for `isQuestionDashcard`?

Reply:
as it can be `isVirtualDashcard` as well afaik